test/topology_experimental_raft/test_raft_voters.py

```diff
@@ -0,0 +1,90 @@
+#
+# Copyright (C) 2024-present ScyllaDB
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+#
+
+import asyncio
+import logging
+import pytest
+
+from cassandra.policies import WhiteListRoundRobinPolicy
+
+from test.pylib.manager_client import ManagerClient
+from test.pylib.rest_client import read_barrier
+from test.topology.conftest import cluster_con
+
+
+@pytest.mark.asyncio
+@pytest.mark.xfail(reason='issue #18793')
+# Make sure the algorithm works with different number of nodes
+@pytest.mark.parametrize('num_nodes', [1, 3])
+async def test_raft_voters_multidc(manager: ManagerClient, num_nodes: int):
+    """
+    Test the basic functionality of limited voters in a multi-DC cluster.
+
+    All DCs should now have the same number of voters so the majority shouldn't be lost when one DC out of 3 goes out,
+    even if the DC is disproportionally larger than the other two.
+
+    Arrange:
+        - create 2 smaller DCs and one large DC
+        - the sum of nodes in the smaller DCs is equal to the number of nodes of the large DC
+    Act:
+        - kill all the nodes in the large DC
+        - this would cause the loss of majority in the cluster without the limited voters feature
+    Assert:
+        - test the topology didn't lose the majority by sending a read barrier request to one of the smaller DCs
```

Comment:
nit: "topology didn't lose the majority" -> "group 0 didn't lose the majority"

We consider the problem of losing majority in group 0. It's true that group 0 members and topology members are the same here, but it's unrelated to the problem.

Reply:
done